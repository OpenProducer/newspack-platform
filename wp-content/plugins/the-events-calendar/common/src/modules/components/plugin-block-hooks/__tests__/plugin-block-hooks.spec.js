/**
 * External Dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal Dependencies
 */
import PluginBlockHooks from '../component';

jest.mock( '@wordpress/editor', () => ( {
	InnerBlocks: 'InnerBlocks',
} ) );
jest.mock( '@wordpress/data', () => ( {
	select: jest.fn( () => ( {
		getBlockTypes: jest.fn( () => ( [
			{ name: 'tribe/event-datetime' },
			{ name: 'tribe/event-pro-recurring' },
			{ name: 'tribe/event-pro-exclusion' },
			{ name: 'tribe/event-cool-container1' },
			{ name: 'tribe/event-cool-container2' },
		] ) ),
	} ) ),
} ) );

describe( 'PluginBlockHooks', () => {
	let props;
	beforeEach( () => {
		props = {
			plugins: [ 'events', 'events-pro', 'events-cool' ],
			pluginTemplates: {
				'events': [
					[ 'tribe/event-datetime', {}],
				],
				'events-pro': [
					[ 'tribe/event-pro-recurring', {}],
					[ 'tribe/event-pro-exclusion', {}],
				],
			},
		};
	} );

	test( 'should match snapshot', () => {
		const component = renderer.create(
			<PluginBlockHooks { ...props } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
	test( 'should only render events templates', () => {
		props.plugins = props.plugins.filter( plugin => plugin === 'events' );
		const component = renderer.create(
			<PluginBlockHooks { ...props } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
	test( 'should only render events-pro templates', () => {
		props.plugins = props.plugins.filter( plugin => plugin === 'events-pro' );
		const component = renderer.create(
			<PluginBlockHooks { ...props } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
	test( 'should not hook in unregistered blocks', () => {
		props.pluginTemplates.events.push( [ 'i-dont-exist', {}] );
		const component = renderer.create(
			<PluginBlockHooks { ...props } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
	test( 'should recursively hook blocks', () => {
		delete props.pluginTemplates.events;
		delete props.pluginTemplates[ 'events-pro' ];
		props.pluginTemplates[ 'events-cool' ] = [
			[ 'tribe/event-pro-recurring', {}],
			[ 'tribe/event-cool-container1', {}, [
				[ 'tribe/event-pro-recurring', {}],
				[ 'tribe/event-cool-container2', {}, [
					[ 'tribe/event-pro-recurring', {}],
					[ 'dont-register-me', {}],
				] ],
				[ 'dont-register-2me', {}],
			],
			[ 'dont-register-me', {}],
			],
		];
		const component = renderer.create(
			<PluginBlockHooks { ...props } />
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
