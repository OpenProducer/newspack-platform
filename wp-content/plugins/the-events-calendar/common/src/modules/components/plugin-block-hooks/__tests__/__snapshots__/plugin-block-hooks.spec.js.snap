// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PluginBlockHooks should match snapshot 1`] = `
<div
  className="tribe-common__plugin-block-hook"
>
  <InnerBlocks
    template={
      Array [
        Array [
          "tribe/event-datetime",
          Object {},
        ],
        Array [
          "tribe/event-pro-recurring",
          Object {},
        ],
        Array [
          "tribe/event-pro-exclusion",
          Object {},
        ],
      ]
    }
    templateInsertUpdatesSelection={false}
  />
</div>
`;

exports[`PluginBlockHooks should not hook in unregistered blocks 1`] = `
<div
  className="tribe-common__plugin-block-hook"
>
  <InnerBlocks
    template={
      Array [
        Array [
          "tribe/event-datetime",
          Object {},
        ],
        Array [
          "tribe/event-pro-recurring",
          Object {},
        ],
        Array [
          "tribe/event-pro-exclusion",
          Object {},
        ],
      ]
    }
    templateInsertUpdatesSelection={false}
  />
</div>
`;

exports[`PluginBlockHooks should only render events templates 1`] = `
<div
  className="tribe-common__plugin-block-hook"
>
  <InnerBlocks
    template={
      Array [
        Array [
          "tribe/event-datetime",
          Object {},
        ],
      ]
    }
    templateInsertUpdatesSelection={false}
  />
</div>
`;

exports[`PluginBlockHooks should only render events-pro templates 1`] = `
<div
  className="tribe-common__plugin-block-hook"
>
  <InnerBlocks
    template={
      Array [
        Array [
          "tribe/event-pro-recurring",
          Object {},
        ],
        Array [
          "tribe/event-pro-exclusion",
          Object {},
        ],
      ]
    }
    templateInsertUpdatesSelection={false}
  />
</div>
`;

exports[`PluginBlockHooks should recursively hook blocks 1`] = `
<div
  className="tribe-common__plugin-block-hook"
>
  <InnerBlocks
    template={
      Array [
        Array [
          "tribe/event-pro-recurring",
          Object {},
        ],
        Array [
          "tribe/event-cool-container1",
          Object {},
          Array [
            Array [
              "tribe/event-pro-recurring",
              Object {},
            ],
            Array [
              "tribe/event-cool-container2",
              Object {},
              Array [
                Array [
                  "tribe/event-pro-recurring",
                  Object {},
                ],
              ],
            ],
          ],
        ],
      ]
    }
    templateInsertUpdatesSelection={false}
  />
</div>
`;
